starfish/codebook.py

```diff
@@ -356,13 +358,65 @@ def to_json(self, filename: str) -> None:
         with open(filename, 'w') as f:
             json.dump(code_array, f)
 
-    def decode_euclidean(self, intensities: IntensityTable) -> IntensityTable:
+    @staticmethod
+    def _unit_normalize(array: xr.DataArray) -> xr.DataArray:
+        """unit normalize each feature of array"""
+        total_intensities = array.sum(dim=[Indices.CH.value, Indices.HYB.value])
+        array = array / total_intensities
```

Comment:
What if the array is all zeros?  :)

Reply:
The function will propagate np.NaN throughout the array. The lines below your comment deal with the result. For unit normalization, which I mistakenly thought that MERFISH was using, this is a really easy fix -- I ensure the feature equals one by filling each (ch, r) combination with the correct fraction of `1`: 

```
        # if a code is all zero, the unit intensity should be spread across the channel
        partitioned_intensity = 1 / np.product(array.shape[1:])
        array = array.fillna(partitioned_intensity)
```

I will need to think about what to do in the 2-norm case.